daemon/cmd/daemon_main.go

```diff
@@ -210,9 +210,6 @@ func init() {
 	flags.Bool(option.AnnotateK8sNode, defaults.AnnotateK8sNode, "Annotate Kubernetes node")
 	option.BindEnv(option.AnnotateK8sNode)
 
-	flags.Bool(option.BlacklistConflictingRoutes, defaults.BlacklistConflictingRoutes, "Don't blacklist IP allocations conflicting with local non-cilium routes")
-	option.BindEnv(option.BlacklistConflictingRoutes)
```

Comment:
Usually we would step this out across two releases: One that marks it as deprecated (potentially raising a warning message to notify the user), then one to delete. That way if someone hardcodes their arguments to include this option, they'll have one upgrade cycle to account for the removed option & remove it.

I think the rest of the code is largely fine, but maybe we should just mark this deprecated here? /cc @aanm

Reply:
I had a discussion around this with @tgraf on slack. He said we've considered to use the regular deprecation but given that we are not going to provide similar functionality that a user can switch to, we can also remove it right away with a note in the release notes.

Reply:
@fristonio We need to mark is as deprecated first otherwise it can break initialization of 1.9 cilium agent for users that use flags from 1.8 since Cilium does not accept undefined flags.

Reply:
So is it like, we keep the flag marking it as deprecated but remove the functionality behind it?

Reply:
Correct. We did something similar for the container runtime flags. https://github.com/cilium/cilium/pull/9447